### PR TITLE
GH-38599: [Docs] Update Headers

### LINKED ISSUE #38599: [Docs][Specifications] Is this intentional or a typo?
### Describe the bug, including details regarding any error messages, version, and platform.

The document for `Device Stream Interface` shows the heading at `Level 1`, but it is not included in `Section Navigation`.

Is this intentional? If it's intentional, shouldn't the heading level for this content be `Level 2`? If it's not intentional, is it okay to adjust it to be positioned independently between `The Arrow C Device data interface` and `Arrow Flight RPC`?

image

* [Specifications](https://arrow.apache.org/docs/dev/format/index.html) 
* [Device Stream Interface](https://arrow.apache.org/docs/dev/format/CDeviceDataInterface.html#device-stream-interface)

### Component(s)

Documentation

### PR DESCRIPTION
**Pull request by llama90**
### Rationale for this change

Noticed wrong section headings on the web doc and proposing a fix.

### Are these changes tested?

Yes. Built and verified the documentation locally.

image

### Are there any user-facing changes?

No.
* Closes: #38599

### CODE REVIEW

**Comment by kou** [comment]
@github-actions crossbow submit preview-docs

File: docs/source/format/CDeviceDataInterface.rst [hunk 1]
```diff
@@ -627,38 +626,38 @@ streaming source of Arrow arrays. It has the following fields:
...
--------------
+'''''''''''''
 
 The stream source is not assumed to be thread-safe. Consumers wanting to
 call ``get_next`` from several threads should ensure those calls are
 serialized.
 
 Interoperability with other interchange formats
-===============================================
+-----------------------------------------------
```
Comment (pitrou) [inline docs/source/format/CDeviceDataInterface.rst:649]: This should be kept as-is as it's a top level section in this page.

File: docs/source/format/CDeviceDataInterface.rst [hunk 2]
```diff
@@ -627,38 +626,38 @@ streaming source of Arrow arrays. It has the following fields:
...
-shape / types of the data alongside the ``ArrowDeviceArray``, users 
-should utilize the existing ``ArrowSchema`` structure to pass any data 
+device data buffers that are being shared. Rather than store the
+shape / types of the data alongside the ``ArrowDeviceArray``, users
+should utilize the existing ``ArrowSchema`` structure to pass any data
 type and shape information.
 
 Updating this specification
-===========================
+---------------------------
```
Comment (pitrou) [inline docs/source/format/CDeviceDataInterface.rst:660]: Same here.

**Review by pitrou — CHANGES_REQUESTED** [review]

File: docs/source/format/CDeviceDataInterface.rst (hunk 1 above)
Comment (llama90) [reply docs/source/format/CDeviceDataInterface.rst]: Thank you for review. Fixed it.

File: docs/source/format/CDeviceDataInterface.rst (hunk 2 above)
Comment (llama90) [reply docs/source/format/CDeviceDataInterface.rst]: Fixed it too.

Comment (llama90) [reply docs/source/format/CDeviceDataInterface.rst]: image

**Review by pitrou — APPROVED** [review]

**Pull request merged**